mtags-shared/src/main/scala/scala/meta/internal/pc/Decoration.scala

```diff
@@ -0,0 +1,11 @@
+package scala.meta.internal.pc
+
+import scala.meta.pc.SyntheticDecoration
+
+import org.eclipse.{lsp4j => l}
+
+case class Decoration(
+    range: l.Range,
+    label: String,
+    kind: Int
```

Comment:
I'm not sure what you mean.
Are you thinking about creating hover for synthetic decorations using PC? It probably can be done, but I'm not sure it's worth it if we want to switch to `inlayHints` soon. Also, we can't really use the same mechanism for hover as for `inlayHints/resolve`.
Theoretically, we can do the same thing for hover as we do now with semanticdb, only with PC, but I don't think it's worth the effort

Reply:
Ach, so for inlay hints we will reuse the normal hover? Or add another method for it?

The issue I saw here is that if we wanted to include hover in the future we would need to create a new class altogether. I guess this is not an issue since the hover will be lazy anyway.

Reply:
For inlay hints we would still need different class, because we would have the decoration split into `label parts`, so adding additional field shouldn’t be a problem. We could do this now, but it's a lot of new complexity without any result (this would require changes in the client or caching decorations for hover)

Reply:
We could potentially add some fields with default values later on.

Reply:
Anyway, ideally we wouldn't need to add spearte classes for both, but if that is too much complexity then it's fine.